### PR TITLE
chore(nuxt): bump nuxt dependencies

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Bump nuxt dependencies to resolve this issue:

```
[nitro] [unhandledRejection] pushWarningContext is not a function
```

## Testing

Local tested, playground runs well.

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [ ] integration tests
- [ ] necessary TSDoc comments

### CODE REVIEW

File: packages/nuxt/tsconfig.json [hunk 1]
```diff
@@ -2,5 +2,7 @@
   "extends": "./.nuxt/tsconfig",
   "compilerOptions": {
     "outDir": "dist",
-  }
+    "skipLibCheck": true
```
Comment (charIeszhao) [inline packages/nuxt/tsconfig.json:5]: Why do we skip lib check?

File: packages/nuxt/package.json [hunk 2]
```diff
@@ -30,20 +30,20 @@
...
     "happy-dom": "^15.10.2",
     "lint-staged": "^15.0.0",
-    "nuxt": "^3.12.4",
+    "nuxt": "^3.15.0",
     "prettier": "^3.0.0",
     "typescript": "^5.3.3",
-    "vitest": "^1.6.0",
-    "vue": "^3.4.19"
+    "vitest": "^2.1.8",
+    "vue": "^3.5.13"
```
Comment (charIeszhao) [inline packages/nuxt/package.json:46]: Sync vue and vitest version in other packages

Comment (wangsijie) [reply packages/nuxt/package.json:46]: We can do it later, this PR is for nuxt.

File: packages/nuxt/tsconfig.json (hunk 1 above)
Comment (wangsijie) [reply packages/nuxt/tsconfig.json:5]: This is the the recommended config.

**Review by charIeszhao — APPROVED** [review]

**Pull request merged**